### ISSUE TITLE
Adding a signature to each address

### ISSUE THREAD

**Issue by lane-c-wagner** [lane-c-wagner/ccap#4]
When an address is submitted to a server, we can require that a signature is included, proving that the sender actually owns the address.

**Comment by FrilledShark** [comment]
This would provide protection against what attack vector?

**Comment by danhalliday** [comment]
Good idea. I was also wondering about something like this after reading about the ‘address update’ attack vector.

Perhaps to replace or add an alias, the user (or rather their client software) would have to sign a message using the private key of their existing address.

Then again this might be a real pain if you lose access to a wallet, you could end up with a lot of orphaned aliases!

**Comment by shilch** [comment]
Yeah, this type of signature would be rather useless.  
However, there are two types of signatures that we could build into the protocol:  

1. **DNS Signatures** - A public key is written as a TXT record into the DNS entries. Every address returned by the CCAP has to be signed by the corresponding private key then. This would be a solution to prevent 3rd party servers from sending a wrong address.  

2. **PGP Signatures** - An uploader can sign an address using his PGP key. This would allow a sender to verify the signature if both have exchanged their keys previously.
